pkg/pillar/cmd/msrv/activatecred.go

```diff
@@ -0,0 +1,134 @@
+package msrv
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/go-tpm/legacy/tpm2"
+	"github.com/google/go-tpm/tpmutil"
+	etpm "github.com/lf-edge/eve/pkg/pillar/evetpm"
+)
+
+// ActivateCredTpmParam provides the parameters required to activate the
+// credential using HWTPM EK and HWTPM AIK.
+type ActivateCredTpmParam struct {
+	Ek      string `json:"ek"`      // HWTPM EK public key
+	AikPub  string `json:"aikpub"`  // HWTPM AIK public key
+	AikName string `json:"aikname"` // HWTPM AIK name
+}
+
+// ActivateCredGenerated contains the generated credential and data to sign.
+type ActivateCredGenerated struct {
+	Data   string `json:"data"`
+	Cred   string `json:"cred"`
+	Secret string `json:"secret"`
+}
+
+// ActivateCredActivated contains the activated credential (decrypted secret)
+// and signature of the ActivateCredGenerated.Data.
+type ActivateCredActivated struct {
+	Secret string `json:"secret"`
+	Sig    string `json:"sig"`
+}
+
+// handles the GET request /tmp/activate-credential/, this is used to get the
+// HTPM EK public key, HWTPM AIK public key, HWTPM AIK name.
+func getActivateCredntialParams() ([]byte, []byte, []byte, error) {
+	rw, err := tpm2.OpenTPM(etpm.TpmDevicePath)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	defer rw.Close()
+
+	ekPub, _, _, err := tpm2.ReadPublic(rw, etpm.TpmEKHdl)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	ekPubByte, err := ekPub.Encode()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var aikName tpmutil.U16Bytes
+	aikPub, aikName, _, err := tpm2.ReadPublic(rw, etpm.TpmAIKHdl)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	aikPubByte, err := aikPub.Encode()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	aikNameMarshaled := &bytes.Buffer{}
+	if err := aikName.TPMMarshal(aikNameMarshaled); err != nil {
+		return nil, nil, nil, err
+	}
+
+	return ekPubByte, aikPubByte, aikNameMarshaled.Bytes(), nil
+}
+
+// handles the POST request /tmp/activate-credential/, this is used to activate
+// the credential (decrypt the secret) using HWTPM EK and HWTPM AIK.
```

Comment:
Are we using HWTPM elsewhere? I wonder whether it is better to just use TPM to refer to it, since that avoids confusion around dTPM vs fTPM and other names which refer to different implementations of what EVE/Linux sees as "hardware".